### PR TITLE
MOEN-18794-PluginBase: Updated initialization method with sdkState

### PR DESCRIPTION
**Pull request by RakshithaAcharya**
Updated Initialization method with SDKState. Instead of passing bool for sdkState, now initialization method accepts parameter of type MoEngageSDKState

### CODE REVIEW

**Comment by RakshithaAcharya** [comment]
@RakshithaAcharya update the version

File: Example/Podfile.lock [hunk 1]
```diff
@@ -16,7 +16,7 @@ PODS:
   - MoEngagePluginInbox (1.0.0):
     - MoEngageInbox (< 1.4.0, >= 1.3.0)
     - MoEngagePluginBase (< 3.1.0, >= 3.0.0)
-  - MoEngageRichNotification (6.3.1):
+  - MoEngageRichNotification (6.3.2):
```
Comment (deepa-iOS-SDK) [inline Example/Podfile.lock:19]: @RakshithaAcharya Any reason for committing podfile.lock changes?

File: MoEngagePluginBase/MoEngagePlugin.swift [hunk 2]
```diff
@@ -53,32 +59,39 @@ import MoEngageInApps
...
-        setDelegates(identifier: sdkConfig.identifier)
+        commonSetUp(identifier: sdkConfig.identifier)
     }
     
     @objc public func trackPluginInfo(_ pluginType: String, version: String) {
         let integrationInfo = MOIntegrationInfo(pluginType: pluginType, version: version)
         MOCoreIntegrator.sharedInstance.addIntergrationInfo(info: integrationInfo)
     }
     
+    private func commonSetUp(identifier: String) {
```
Comment (deepa-iOS-SDK) [inline MoEngagePluginBase/MoEngagePlugin.swift:88]: @RakshithaAcharya Was there any reason for creating another function for setDelegates here?

File: MoEngagePluginBase/MoEngagePlugin.swift [hunk 3]
```diff
@@ -53,32 +59,39 @@ import MoEngageInApps
...
     }
     
+    private func initializeMoEngageSecondaryInstance(sdkConfig: MOSDKConfig, sdkState: MoEngageSDKState) {
+       #if DEBUG
+         MoEngage.sharedInstance().initializeTestInstance(with: sdkConfig, andSDKState: sdkState)
+      #else
+        MoEngage.sharedInstance().initializeLiveInstance(with: sdkConfig, andSDKState: sdkState)
+      #endif
+        
+     commonSetUp(identifier: sdkConfig.identifier)
```
Comment (deepa-iOS-SDK) [inline MoEngagePluginBase/MoEngagePlugin.swift:69]: check for indentation here @RakshithaAcharya

File: MoEngagePluginBase/MoEngagePlugin.swift [hunk 4]
```diff
@@ -27,6 +25,16 @@ import MoEngageInApps
...
     }
     
+    private func initializeMoEngageDefaultInstance(sdkConfig: MOSDKConfig, sdkState: MoEngageSDKState) {
+     #if DEBUG
+            MoEngage.sharedInstance().initializeDefaultTestInstance(with: sdkConfig, andSDKState: sdkState)
+     #else
+            MoEngage.sharedInstance().initializeDefaultLiveInstance(with: sdkConfig, andSDKState: sdkState)
+    #endif
+        
+    commonSetUp(identifier: sdkConfig.identifier)
```
Comment (deepa-iOS-SDK) [inline MoEngagePluginBase/MoEngagePlugin.swift:35]: @RakshithaAcharya Check for indentation here

File: Example/Podfile.lock (hunk 1 above)
Comment (RakshithaAcharya) [reply Example/Podfile.lock:19]: @deepa-iOS-SDK This is part of sample app. This Keeps the track of version in which last testing was done.

File: MoEngagePluginBase/MoEngagePlugin.swift (hunk 2 above)
Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePlugin.swift:88]: @deepa-iOS-SDK Moving forward if there is any code that should be called post initialization(with different initialization methods), will go inside crommonsetup.

File: MoEngagePluginBase/MoEngagePlugin.swift (hunk 3 above)
Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePlugin.swift]: done

File: MoEngagePluginBase/MoEngagePlugin.swift (hunk 4 above)
Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePlugin.swift]: done

**Review by umangmoe — APPROVED** [review]

**Pull request merged**